File: python/src/n0226_invert_binary_tree_test.py
from n0226_invert_binary_tree import invert_tree, invert_tree_recursive
from utils.tree import Tree


def test_invert_binary_tree_example_one() -> None:
    tree = Tree([4, 2, 7, 1, 3, 6, 9])
    tree.root = invert_tree(tree.root)
    assert tree.to_list() == [4, 7, 2, 9, 6, 3, 1]


def test_invert_binary_tree_example_two() -> None:
    tree = Tree([2, 1, 3])
    tree.root = invert_tree(tree.root)
    assert tree.to_list() == [2, 3, 1]


def test_invert_binary_tree_example_three() -> None:
    tree = Tree([])
    assert tree.root is None
    tree.root = invert_tree(tree.root)
    assert tree.root is None


def test_invert_binary_tree_recursive_example_one() -> None:
    tree = Tree([4, 2, 7, 1, 3, 6, 9])
    tree.root = invert_tree_recursive(tree.root)
    assert tree.to_list() == [4, 7, 2, 9, 6, 3, 1]


def test_invert_binary_tree_recursive_example_two() -> None:
    tree = Tree([2, 1, 3])
    tree.root = invert_tree_recursive(tree.root)
    assert tree.to_list() == [2, 3, 1]


def test_invert_binary_tree_recursive_example_three() -> None:
    tree = Tree([])
    assert tree.root is None
    tree.root = invert_tree_recursive(tree.root)
    assert tree.root is None

File: rust/src/n0226_invert_binary_tree.test.rs
use crate::n0226_invert_binary_tree::*;
use crate::utils::tree::Tree;

#[test]
fn test_invert_binary_tree_example_one() {
    let mut tree = Tree::new([
        Some(4),
        Some(2),
        Some(7),
        Some(1),
        Some(3),
        Some(6),
        Some(9),
    ]);
    tree.root = invert_binary_tree(tree.root);
    assert_eq!(tree.to_vec(), [4, 7, 2, 9, 6, 3, 1]);
}

#[test]
fn test_invert_binary_tree_example_two() {
    let mut tree = Tree::new([Some(2), Some(1), Some(3)]);
    tree.root = invert_binary_tree(tree.root);
    assert_eq!(tree.to_vec(), [2, 3, 1]);
}

#[test]
fn test_invert_binary_tree_example_three() {
    let mut tree = Tree::new([]);
    tree.root = invert_binary_tree(tree.root);
    assert_eq!(tree.to_vec(), []);
}

#[test]
fn test_invert_binary_tree_recursive_example_one() {
    let mut tree = Tree::new([
        Some(4),
        Some(2),
        Some(7),
        Some(1),
        Some(3),
        Some(6),
        Some(9),
    ]);
    tree.root = invert_binary_tree_recursive(tree.root);
    assert_eq!(tree.to_vec(), [4, 7, 2, 9, 6, 3, 1]);
}

#[test]
fn test_invert_binary_tree_recursive_example_two() {
    let mut tree = Tree::new([Some(2), Some(1), Some(3)]);
    tree.root = invert_binary_tree_recursive(tree.root);
    assert_eq!(tree.to_vec(), [2, 3, 1]);
}

#[test]
fn test_invert_binary_tree_recursive_example_three() {
    let mut tree = Tree::new([]);
    tree.root = invert_binary_tree_recursive(tree.root);
    assert_eq!(tree.to_vec(), []);
}
